src/axolotl/state.rs

```diff
@@ -144,22 +143,25 @@ impl <T:Axolotl> ReceiveChain<T> {
         let (_,ref message_key) = self.message_keys[message_key_index];
         let expected_mac = axolotl_impl.authenticate_message(message, message_key, sender_identity, receiver_identity);
         if expected_mac == mac {
-            axolotl_impl.decrypt_message(&message_key, &message.ciphertext)
+            axolotl_impl.with_decoded_ciphertext(message, |ciphertext| {
```

Comment:
Is this a common pattern in rust? It seems like something they'd want to solve more elegantly.